tests/kani/Invariant/percentage.rs

```diff
@@ -0,0 +1,33 @@
+// Copyright Kani Contributors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+//! Check that the `Invariant` implementation behaves as expected when used on a
+//! custom type.
+
+extern crate kani;
+use kani::Invariant;
+
+// We use the default `Arbitrary` implementation, which allows values that
+// shouldn't be considered safe for the `Percentage` type.
+#[derive(kani::Arbitrary)]
+struct Percentage(u8);
```

Comment:
Can I suggest we enhance this test a bit? Maybe add a few functions to this type, like `try_new`, and a `value` method to show that users can only violate the safety invariant by using unsafe code.

Reply:
Done. Also added an `increase` method which uses `try_new` which is demonstrated to be safe w.r.t. the invariant through another harness.